src/EventLogExpert/Components/EventTable.razor

```diff
@@ -3,91 +3,91 @@
+            <th class="source" hidden="@(IsColumnHidden(ColumnName.Source))">
+                Source
                     @if (EventTableState.Value.OrderBy == ColumnName.Source)
-                    {
+                {
                         <span class="menu-toggle" data-rotate="@EventTableState.Value.IsDescending.ToString().ToLower()" @onclick="ToggleSorting">
-                            <i class="bi bi-caret-up"></i>
-                        </span>
-                    }
-                </th>
-                <th class="id" hidden="@(IsColumnHidden(ColumnName.EventId))">
-                    Event ID
+                        <i class="bi bi-caret-up"></i>
+                    </span>
+                }
+            </th>
+            <th class="id" hidden="@(IsColumnHidden(ColumnName.EventId))">
+                Event ID
                     @if (EventTableState.Value.OrderBy == ColumnName.EventId)
-                    {
+                {
                         <span class="menu-toggle" data-rotate="@EventTableState.Value.IsDescending.ToString().ToLower()" @onclick="ToggleSorting">
-                            <i class="bi bi-caret-up"></i>
-                        </span>
-                    }
-                </th>
-                <th class="task" hidden="@(IsColumnHidden(ColumnName.TaskCategory))">
-                    Task Category
+                        <i class="bi bi-caret-up"></i>
+                    </span>
+                }
+            </th>
+            <th class="task" hidden="@(IsColumnHidden(ColumnName.TaskCategory))">
+                Task Category
                     @if (EventTableState.Value.OrderBy == ColumnName.TaskCategory)
-                    {
+                {
                         <span class="menu-toggle" data-rotate="@EventTableState.Value.IsDescending.ToString().ToLower()" @onclick="ToggleSorting">
-                            <i class="bi bi-caret-up"></i>
-                        </span>
-                    }
-                </th>
-                <th class="description">Description</th>
-            </tr>
-            </thead>
-            <tbody @oncontextmenu="InvokeContextMenu">
-            <Virtualize Items="table.DisplayedEvents" Context="evt">
-                <tr @key="@($"{table.Id}_{evt.RecordId}")" class="@GetCss(evt)" @onfocus="() => SelectEvent(evt)" tabindex="0">
+                        <i class="bi bi-caret-up"></i>
+                    </span>
+                }
+            </th>
+            <th class="description">Description</th>
+        </tr>
+        </thead>
+        <tbody @oncontextmenu="InvokeContextMenu">
+        @if (EventTableState.Value.ActiveTableId is not null)
+        {
+            <Virtualize Items="EventTableState.Value.EventTables.First(x => x.Id == EventTableState.Value.ActiveTableId).DisplayedEvents" Context="evt">
+                <tr @key="@($"{evt.LogName}_{evt.RecordId}")" class="@GetCss(evt)" @onfocus="() => SelectEvent(evt)" tabindex="0">
```

Comment:
```suggestion
                <tr @key="@($"{evt.OwningLog}_{evt.RecordId}")" class="@GetCss(evt)" @onfocus="() => SelectEvent(evt)" tabindex="0">
```

This is still vulnerable to duplicate key unless we use OwningLog. LogName is the log name such as "Application" or "System". If you open two Application logs with overlapping RecordId values, you will get a duplicate key. OwningLog is the file name for files and the log name for live logs. Since you can't open the same file twice or the same live log twice in one instance of EventLogExpert, a duplicate key is not possible if we use OwningLog.